fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -405,6 +408,34 @@ public PlanFragment visitPhysicalFilter(PhysicalFilter<Plan> filter, PlanTransla
         return inputFragment;
     }
 
+    @Override
+    public PlanFragment visitPhysicalLimit(PhysicalLimit<Plan> physicalLimit, PlanTranslatorContext context) {
+        PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
+        PlanNode child = inputFragment.getPlanRoot();
+        if (child instanceof OlapScanNode) {
+            child.setLimit(physicalLimit.getLimit() + physicalLimit.getOffset());
+            return inputFragment;
+        }
+
+        if (child instanceof SortNode) {
+            ((SortNode) child).setOffset(physicalLimit.getOffset());
+            child.setLimit(physicalLimit.getLimit());
+            inputFragment.getChildren().forEach(fragment -> {
+                PlanNode root = fragment.getPlanRoot();
+                if (root instanceof SortNode) {
+                    root.setLimit(physicalLimit.getLimit() + physicalLimit.getOffset());
+                }
+            });
+            return inputFragment;
+        }
+
+        if (child instanceof AggregationNode) {
```

Comment:
except sort node, all other node should use same logical to do limit merge, right?

Reply:
I think so, too.
The sort node + limit = TopN. other plan + limit can simple setLimit.